File: Homework3/thermostat/thermostat.java
package thermostat;

public class thermostat
{
    private int temp;
    private tempType type;
    private fanSpeed speed;

    public thermostat() {
        setTemp(75);
        setTempType(tempType.F);
        setFanSpeed(fanSpeed.AUTO);
    }

    public thermostat(int _temp, tempType _type, fanSpeed _speed) {
        setTemp(_temp);
        setTempType(_type);
        setFanSpeed(_speed);
    }

    public int getTemp() {
        return temp;
    }

    public tempType getTempType() {
        return type;
    }

    public fanSpeed getFanSpeed() {
        return speed;
    }

    public void setTemp(int _temp) {
        if(_temp > 0 && _temp < 100)
            temp = _temp;
        else {
            throw new IllegalArgumentException("Temp was " + _temp + ". It should be between 0 and 100.");
        }
    }

    public void setTempType(tempType _type) {
        type = _type;
    }

    public void setFanSpeed(fanSpeed _speed) {
        speed = _speed;
    }
}

File: Homework4/src/Thermostat.cpp
#include "Thermostat.h"
#include <stdexcept>

namespace thermostat{
    
    Thermostat::Thermostat() 
        : temp(75), type(tempType::F), speed(fanSpeed::AUTO) {}
    
    Thermostat::Thermostat(int _temp, tempType _type, fanSpeed _speed)
        : temp(_temp > 0 && _temp < 100 ? _temp : 0), type(_type), speed(_speed)
    {
        if (temp <= 0) {
            throw std::invalid_argument("Temp was " + std::to_string(temp) + ". It should be between 0 and 100.");
        }
    }

    int Thermostat::getTemp() const {
        return temp;
    }

    tempType Thermostat::getTempType() const {
        return type;
    }

    fanSpeed Thermostat::getFanSpeed() const {
        return speed;
    }

    void Thermostat::setTemp(int _temp) {
        if(_temp > 0 && _temp < 100)
            temp = _temp;
        else
            throw std::invalid_argument("Temp was " + std::to_string(_temp) + ". It should be between 0 and 100.");
    }

    void Thermostat::setTempType(tempType _type) {
        type = _type;
    }

    void Thermostat::setFanSpeed(fanSpeed _speed) {
        speed = _speed;
    }
}
